code/game/objects/effects/spawners/random/maintenance.dm

```diff
@@ -19,6 +19,13 @@
 	. += span_info("This spawner has an effective loot count of [get_effective_lootcount()].")
 
 /obj/effect/spawner/random/maintenance/Initialize(mapload)
+	if(!istype(src, /obj/effect/spawner/random/maintenance/no_decals) && is_space_or_openspace(loc))
+		if(mapload)
+			stack_trace("[src] at [AREACOORD(src)] tried to spawn in a space or openspace tile! Mappers please use /obj/effect/spawner/random/maintenance/no_decals instead! Note: the coords might be for a ruin so keep that in mind.")
+		else
+			new /obj/effect/spawner/random/maintenance/no_decals(loc) // if this happens outside of mapload, just spawn a no_decals version and move on
+		return INITIALIZE_HINT_QDEL
```

Comment:
This breaks spawn amount for subtypes, better to runtime on mapload and check for turf validity in can_spawn